WrightTools/_group.py

```diff
@@ -178,21 +179,113 @@ def parent(self):
         except (AssertionError, AttributeError):
             from .collection import Collection
-            name = super().parent.attrs['name']
-            self._parent = Collection(self.filepath, name=name, edit_local=True)
+            key = posixpath.dirname(self.fullpath) + posixpath.sep
+            self._parent = Collection.instances[key]
         finally:
             return self._parent
 
     def close(self):
         """Close the group. Tempfile will be removed, if this is the final reference."""
         if(self.fid.valid > 0):
-            self.__class__.instances.pop(self.fullpath)
-            self.file.flush()
-            self.file.close()
-            if hasattr(self, '_tmpfile'):
-                os.close(self._tmpfile[0])
-                os.remove(self._tmpfile[1])
+            self.__class__.instances.pop(self.fullpath, None)
+            try:
+                self.file.flush()
+                self.file.close()
+            except SystemError:
+                pass
+            finally:
+                if hasattr(self, '_tmpfile'):
+                    os.close(self._tmpfile[0])
+                    os.remove(self._tmpfile[1])
+
+    def copy(self, parent=None, name=None, verbose=True):
+        """Create a copy under parent.
+
+        All children are copied as well.
+
+        Parameters
+        ----------
+        parent : WrightTools Collection (optional)
+            Parent to copy within. If None, copy is created in root of new
+            tempfile. Default is None.
+        name : string (optional)
+            Name of new copy at destination. If None, the current natural
+            name is used. Default is None.
+        verbose : boolean (optional)
+            Toggle talkback. Default is True.
+
+        Returns
+        -------
+        Group
+            Created copy.
+        """
+        if name is None:
+            name = self.natural_name
+        if parent is None:
+            new = Group()
+            # attrs
+            for k, v in self.attrs.items():
+                new.attrs[k] = v
+            new.natural_name = name
+            # children
+            for k in self.keys():
+                self[k].copy(new)
+            new.flush()
+            p = new.filepath
+            from ._open import open as wt_open
+            new = wt_open(p)
+        else:
+            # copy
+            self.file.copy(self.name, parent, name=name)
+            if 'item_names' in parent.attrs.keys():
+                new = parent.item_names + (name,)
+                parent.attrs['item_names'] = np.array(new, dtype='S')
+            parent._update_natural_namespace()
+            new = parent[name]
+        # finish
+        if verbose:
+            print('{0} copied to {1}'.format(self.fullpath, new.fullpath))
+        return new
 
     def flush(self):
         """Ensure contents are written to file."""
         self.file.flush()
+
+    def save(self, filepath=None, verbose=True):
```

Comment:
I would like overwrite protection (i.e. `overwrite=False` kwarg, exception/warning when the save operation fails/destroys a previous copy)